src/main/java/io/electrum/airtime/api/IPurchaseResource.java

```diff
@@ -30,6 +31,29 @@ void purchasePurchase(
          UriInfo uriInfo,
          HttpServletRequest httpServletRequest);
 
+   /**
+    *
+    * @since v5.15.0
+    *
+    * @param body
+    * @param securityContext
+    * @param request
+    * @param httpHeaders
+    * @param asyncResponse
+    * @param uriInfo
+    * @param httpServletRequest
+    */
+   default void trialPurchase(
+         PurchaseRequest body,
+         SecurityContext securityContext,
+         Request request,
+         HttpHeaders httpHeaders,
+         AsyncResponse asyncResponse,
+         UriInfo uriInfo,
+         HttpServletRequest httpServletRequest) {
+      asyncResponse.resume(new ServerErrorException("This operation has not been implemented.", 501));
```

Comment:
```suggestion
      throw new UnsupportedOperationException("This operation has not been implemented.");
```

Both of these will result in an HTTP 501 being returned to the client. But calling `asyncResponse.resume` forces that response without the implementing class having the opportunity to intercept it.

I think throwing the exception provides better transparency to the implementer who can catch it and handle it in some other way if desired.

Reply:
Do you say that both of these will result in a HTTP 501 being returned to the client since the Jetty Server will handle the `UnsupportedOperationException` appropriately?

Since the implementing class with override this method if desired, then they do have the opportunity to change the response to whatever they desire, however this implementation ensures that a HTTP 501 is returned and that the implementing class only needs to implement this method if actually desired.

I'm going to leave this unchanged pending the discussion in [the ppu review](https://github.com/electrumpayments/prepaidutility-service-interface/pull/21#discussion_r481846820)